apps/omg_watcher_info/lib/omg_watcher_info/order_fee_fetcher.ex

```diff
@@ -0,0 +1,66 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherInfo.OrderFeeFetcher do
+  @moduledoc """
+  Handle fetching and formatting of fees for an order
+  """
+
+  alias OMG.Crypto
+  alias OMG.State.Transaction
+  alias OMG.Utils.HttpRPC.Encoding
+  alias OMG.WatcherInfo.HttpRPC.Client
+  alias OMG.WatcherInfo.UtxoSelection
+  alias OMG.WireFormatTypes
+
+  # Note: Hardcoding the tx_type for now
+  @tx_type WireFormatTypes.tx_type_for(:tx_payment_v1)
+  @str_tx_type Integer.to_string(@tx_type)
+
+  @type order_without_fee_amount_t() :: %{
+          owner: Crypto.address_t(),
+          payments: nonempty_list(UtxoSelection.payment_t()),
+          fee: %{currency: Transaction.Payment.currency()},
+          metadata: binary() | nil
+        }
+
+  @doc """
+  Fetch the correct fee amount for the given fee currency from the childchain
+  and adds it to the order map.
+  """
+  @spec add_fee_to_order(order_without_fee_amount_t()) :: {:ok, UtxoSelection.order_t()} | {:error, atom()}
+  def add_fee_to_order(%{fee: %{currency: currency}} = order, url \\ nil) do
+    child_chain_url = url || Application.get_env(:omg_watcher_info, :child_chain_url)
```

Comment:
I think a better pattern would be to inject the child chain URL as a plug on routes that need it. I would like us to start avoiding configuration fetches in this manner. A good pattern is to think about configuration in a hierarchical manner. For example: configuration must come from the top while propagated to the bottom (this pattern avoids shared state problems).

Reply:
I disagree, I don't think accessing configuration values belong at the web/conn/plug level.

Reply:
the problem comes down to shared state. Imagine having a life cycle of a process that is long lived (a websocket connection or a http request would do). 
Your process does several HTTP requests towards a pre-set-configurational value. While the process is still alive this particular configuration value changes. This could make the behavior of your process unpredictable. 
By injecting requirements as early as possible, you fully control the lifecycle of your process.

Reply:
similarly to an OTP process, while attaching it to a supervision tree, you should always pass arguments to it and avoid reading configuration during the process lifecycle

Reply:
thinking further... actually the problem here is our request process scope that's being dragged to deep in the application stack. if we wanted to limit the process scope and have clear separation of responsibilities we should introduce short lived processes that have one and only responsibility: transaction processing